common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java

```diff
@@ -128,6 +138,10 @@ public String toString() {
     return localDateTime.format(PRINT_FORMATTER);
   }
 
+  public String toStingWithLenientFormatter() {
+    return localDateTime.format(PRINT_LENIENT_FORMATTER);
+  }
+
```

Comment:
The use of `Lenient` is a bit misleading as I wrote previously. Also there is a small typo in the method name `toSting` vs `toString`.

Instead of adding a new method we could use `Timestamp#format` passing in the desired formatter. With the right naming for the formatter parameter it would make the intention more clear.

Reply:
Good idea with the `format()` method